### PR TITLE
fix: Fixing CI for TRTLLM HLAPI 

### PR DESCRIPTION
**Pull request by KrishnanPrash**
This PR fixes the CI failures that have been observed when using the TRTLLM HL API. The fixes include:
- Mainly, for the `ScopedTritonServer`, it relies on using `subprocess` and it's relevant functions instead of using `psutils` and having to manually keep track of the process id. 
- Additionally, resolving an issue that was observed when stopping triton instance in CI. Here is the error messaged being displayed:
```
Received signal=15, frame=<frame at 0x79d701106f80, file '/mnt/triton_cli/src/triton_cli/server/server_local.py', line 148, code logs>. Shutting down...
reentrant call inside <_io.BufferedReader name=3>
triton - ERROR - Failed to start server, unknown error.
```
The root cause for this was after `process.terminate()` was called for the underlying tritonserver process, it is followed by a `process.communicate(timeout=...)` which triggers a file I/O operation. Hence, to resolve this we replace `process.communicate()` with `process.wait()` to give the process time to clean up with making an I/O call.

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -136,199 +110,44 @@ def _clear(repo=None):
...
 
-        # TODO: Be more coupled with Triton CLI settings
-        self.host = "localhost"
-        self.port = 8000
-        self.start_timeout = timeout
-        self.proc = None
+    def start(self):
+        self.proc = self.run_server(self.repo, self.mode)
+        self.wait_for_server_ready(
+            # url=self.url_for("v2", "health", "ready"),
```
Comment (rmccorm4) [inline tests/utils.py:129]: remove unused url

**Review by rmccorm4 — APPROVED** [review]
LGTM if pipeline passes. Please follow-up on the other PR that this is merging into afterwards

**Review by krishung5 — APPROVED** [review]
LGTM. qq - does this also fix the GPU OOM issue or is it a separate problem?

**Comment by rmccorm4** [comment]
> LGTM. qq - does this also fix the GPU OOM issue or is it a separate problem?

My PR which uses multiprocessing seems to definitively fix the OOM issue. Results were muddied for a bit because it turned out the gitlab mirror got out of sync and was running old code each time.

**Comment by KrishnanPrash** [comment]
@krishung5 That fix was made as a part of Ryan's branch [[Link](https://github.com/triton-inference-server/triton_cli/tree/kprashanth-version-update)]. IIUC, the solution was building the trtllm engine as a separate child process. Here is the relevant code snippet:
```python
# Run TRT-LLM build in a separate process to make sure it definitely
# cleans up any GPU memory used when done.
p = multiprocessing.Process(
         target=self.__build_trtllm_engine, args=(huggingface_id, engines_path)
)
p.start()
p.join()
```

**Pull request merged**